Data/Collections/Util/Array2DExtension.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System;
+
+namespace Anvil.CSharp.Data
+{
+    //TODO: #104 - Optimize
+    /// <summary>
+    /// A set of helper extension methods for 2D arrays
+    /// </summary>
+    public static class Array2DExtension
+    {
+        /// <summary>
+        /// Gets the value at the specified index or returns the default value if out of range.
+        /// </summary>
+        /// <param name="array">The array to query.</param>
+        /// <param name="x">The first order index</param>
+        /// <param name="y">The second order index</param>
+        /// <typeparam name="T">The type of the array.</typeparam>
+        /// <returns>The value at the specified index or default.</returns>
+        public static T GetElementOrDefaultAt<T>(this T[,] array, int x, int y)
```

Comment:
Would an optional explicit default value be useful? Mostly for primitives I think, so you could say `array.GetElementOrDefaultAt(x, y, -1);`